lib/puppet/provider/mongodb.rb

```diff
@@ -132,9 +132,14 @@ def self.conn_string
 
   def self.db_ismaster
     cmd_ismaster = 'db.isMaster().ismaster'
-    cmd_ismaster = mongorc_file + cmd_ismaster if mongorc_file
     db = 'admin'
-    res = mongo_cmd(db, conn_string, cmd_ismaster).to_s.chomp
+
+    if mongorc_file
+      res = mongo_cmd(db, conn_string, mongorc_file + cmd_ismaster).to_s.chomp
+    end
+    if res.match(/Authentication failed/) or not mongorc_file
```

Comment:
Perhaps you should switch these around because `res` will not be defined if `mongorc_file` is false.